File: Lab 3/src/main/java/lab3/UserInfo.java
package lab3;

import java.io.Serializable;
import java.util.List;
import javax.persistence.*;

@Entity
@Table(name = "users")
public class UserInfo implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    @Column(name = "client_id")
    Integer id = null;

    String username = null;
    String password = null;

    @OneToMany(mappedBy = "user", targetEntity = Result.class, cascade = {CascadeType.ALL}, fetch = FetchType.EAGER)
    List<Result> results = null;

    @OneToMany(mappedBy = "user", targetEntity = Group.class, cascade = {CascadeType.ALL}, fetch = FetchType.LAZY)
    List<Group> groups = null;

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public String getUsername() {
        return username;
    }

    public void setUsername(String username) {
        this.username = username;
    }

    public String getPassword() {
        return password;
    }

    public void setPassword(String password) {
        this.password = password;
    }

    public List<Result> getResults() {
        return results;
    }

    public void setResults(List<Result> results) {
        this.results = results;
    }

    public List<Group> getGroups() {
        return groups;
    }

    public void setGroups(List<Group> groups) {
        this.groups = groups;
    }
}

File: lab4/src/main/kotlin/lab4/model/UserInfo.kt
package lab4.model

import javax.persistence.*

@Entity
@Table(name = "users")
data class UserInfo(@Column(nullable = false, unique = true) var username: String = "",
                    var password: String = "") {

    @Id
    @GeneratedValue
    var id: Long? = null

    var active: Boolean = true

    @ElementCollection(targetClass = Role::class, fetch = FetchType.EAGER)
    @CollectionTable(name = "user_role", joinColumns = [JoinColumn(name = "user_id")])
    @Enumerated(EnumType.STRING)
    var roles: Set<Role>? = null

    @OneToMany(fetch = FetchType.LAZY, cascade = [CascadeType.ALL], mappedBy = "userInfo")
    var results: MutableList<Result>? = null

    var overallDots: Long = 0
    var overallElapsedTime: Long = 0
    var overallGames: Long = 0
    var overallWins: Long = 0
    var score: Long = 0
    var performance: Double = 0.0
}